### DISCUSSION TITLE
Feedback, ideas and issues

### DISCUSSION THREAD

**Discussion by rtrzebinski** [nicymike/serflings#121] [General]
Feedback:

I've been playing this game for 25 years, mainly on Amiga 500, but I also know DOS version (SVGA).

I use macOS and I'm running the remake with Wine - works just fine out of the box.

**I must say it's a very special game for me, and the fact that someone made this remake means a lot to me, thank you so much!**

I really love the improvements made by the remake:

Zoom / HD Resolution - amazing, best thing ever.
Automatic roads - wow.
Tab to see notification.
Esc to close window.

Feature request / ideas:

Few things that come to my mind.

RMB to close the menu window when clicked on it (like in s2)

Recruit per castle / magazine. On a big map, you might have scenario where weapons are in the main castle, where you want not to recruit from (to have population to be used for workers). You have however weapon smiths next to magazines, and it would be helpful to have a "recruit" 1/5/20/100 button in a magazine menu - so solders can be recruited manually per magazine. This UI can be put after transport info. Current option allows global manually recruiting, but if you have 10 magazines with weapon and only one with tools, more precision is needed.

Maps bigger than 8, ideally with unlimited population (or at least a very high limit).
Edit: I've just spent 16h on map 8 and I'm not sure if this is really needed.. 8 is a really big map.
I know max limit is uint16, which might be hard to overcome.

Auto destruction of empty mines / stone cutters.

Localize buildings. In the menu where buildings ready/requested are displayed - click on the building to select it on the map (same way as notifications are selecting a place).
Case - I planned multiple mines, never connected them with a road, so they are waiting for the time when needed. Now I need a gold mine, but where is it?

Stop working toggle per building. No explanation needed I guess, especially useful with mines.

Option to hide mission and game goals.

This might be tricky UI wise but - ability to queue tools production per specific tool smith.

Sounds when zoomed out. When I zoom out the game max, on a high res screen, sounds are played from all visible building, making it very noisy. It might not be easy to implement but here's the idea - only play these sounds on closer zoom levels. When zoomed out play only music + nature sounds (birds, wind).

Building priorities. Ability to select a construction site as "high priority" so resources will be directed towards this construction site faster than to others. Perhaps also high priority workers receiving resources faster.

When deleting a flag with a connected road(s), delete these road(s) as well (preferably with a confirmation, like when destroying a building).

Potential issues:

Not much here really, please let me know if any deserves a ticket, I'll create it.

Saving game without name should result in "no name" game, now it's empty.

The "yop" / "nono" / "new message" sounds feel a bit too quiet compared to other sounds in the game. (might be because of high res = a lot of sounds)

Exiting options menu closes options window, it should go one lvl up instead I believe.

Stone cutter can not see stones partially hidden behind a building (perhaps that is a bug in the original game as well). Same for some trees never cut etc.
Edit: I read in another thread this is a know issue.

**Comment by rtrzebinski** [comment]
When space bar is pressed yellow building symbols are visible. How about displaying some info on construction sites? Especially useful would be displaying a main type (just a color perhaps) construction that have not started yet. Case: I made a lot of mines in advance (not connected with roads) and now I don't know where are these gold mines I need to connect. Again, important on a very big map.

**Comment by Darmorque** [comment]
I'd say few "for" and few "against" to the aforementioned list of ideas. Overall I'm all in for all possible quality of life improvements. The whole Serflings project aims at making the game more friendly, more functional, UI being more informative, convenient, etc. The overall experience gets more contemporary, despite huge load of nostalgia and the game being possibly faithful to original title. I love all those changes, no doubt.

On the other hand I'd be causiously against the ability of micromanagement in single, particular buildings. It's perfectly understandable what's the idea behind such improvements BUT "The Settlers" was always the kind of puzzle game of sorts. It's about managing this ever-growing mechanism from the very top, and only from there. I mean: your decisions taken at the top of every production chain or any other pyramid of mutualy-dependant game mechanics, have their effect at the very bottom - at the level of single workers, soldiers, buildings, etc, all without your direct influence. You plan your moves, you execute them and then observe the results of hopefly right choices.

The fact that you cannot solve single problematic issues by altering production priority of single toolmaker, or stop a single mine, actually increases the depth of skillfull management of the whole kingdom, foreseeing potential problems quite early, before they arise, etc. I always really liked this game philosophy. You can't directly control your soldiers, the same with the citizens, and I feel that those little tweaking possibilities of single producion buildings would be somehow against the spirit of "Think globaly and let the details catch you off guard later". ;)

I mean - not being able to fix the results of your prevoius mistakes and oversights on the go makes them actually more challenging to cope with their influence on the whole economy. I'll never forget my soldier being one second late to the newly build castle, before the enemy knight being one second faster to his one on the other side, burning most of my mining business at the same time in one go. My fault, the builder had to wait a bit too long for the last missing piece of stone. That would never happen, had I got the ability to micromanage the stone production or transport process. No consequences of wrong previous chioces too. And that's the lovely process of learning. Thanks for the attention. ;)

**Comment by rtrzebinski** [comment]
@Darmorque Thank you for the input, I understand what you mean and it makes perfect sense.

This is how I see it - there are 2 types of changes - those altering the gameplay (for instance "micromanagement in single, particular buildings") and those not altering it (for instance higher resolution, auto roads construction).

For the first, more controversial type I believe the game could:
- make them optional
- disable by default

This way everyone is happy - conservative players will just enjoy the original game, while for some other players it might be fun to explore the game a bit differently - I'd personally like some experimentation.

@nicymike I have more ideas in the meantime. They might not be sensible or even possible to implement, but since this is the discussion I'll just drop them here, maybe you/community will like some:

Zoom the map in and out with a mouse scroll. Currently on a huge map only a tiny fraction of the country is visible. Also bigger map window - for the same reason.

Fog of war. This IMO would make the game dramatically different in a good way - you don't know where the enemy is, you can't prepare in advance just from one side, I'd love to try that. Also you don't know where mountains and other resources are. Unsure if it is possible to implement FOG for the computer UI as well - might be challenging.

Send multiple geologists in one click. Arrows to increase / decrease number of geologists sent, with default set to 1.

Set military building crew manually for each building (override the global setting with a local setting per building).

Fix: save game not keeping some settings: map, stats. Might not be backward compatible, in which case it's probably not worth it.

Change knights (so weakest knights go back to the warehouses to receive more training) per military building.

Alternative UI themes - winter for instance (here I'm dreaming). Perhaps AI could generate some.

Hunter building / worker and animals respawning in the forest (here I'm even more dreaming :D ). It would not alter game balance too much as it's just another food source, similar to the fisherman.

**Comment by nicymike** [comment]
Thank you for your interest and your ideas. You may create new issues (as you already did). Just remember to create a single issue for every single idea or bug.

I don't have much time right now to work on Serflings but I don't plan to abandon the game. 🙂

> Maps bigger than 8, ideally with unlimited population (or at least a very high limit). Edit: I've just spent 16h on map 8 and I'm not sure if this is really needed.. 8 is a really big map. I know max limit is uint16, which might be hard to overcome.

If you enable preview mode you may play with a map size up to 11 and increased max settlers and buildings. But be warned: You will run into game bugs because the game is not ready to handle the increased max number of settlers yet.

> Localize buildings. In the menu where buildings ready/requested are displayed - click on the building to select it on the map (same way as notifications are selecting a place). Case - I planned multiple mines, never connected them with a road, so they are waiting for the time when needed. Now I need a gold mine, but where is it?

There is a similar feature already included in Serflings because it is also a feature of the original game. Just do a special click or middle click on the button for buildings inside the map window. You will get the option to select a building type.

![image](https://github.com/nicymike/serflings/assets/19415000/78fe5f0e-b858-4ba1-a58c-bb7b8992249a)

> Option to hide mission and game goals.

This setting can be found in options window at game options.

![image](https://github.com/nicymike/serflings/assets/19415000/e8a75d25-ebca-4aa7-aea9-a4b3cbb022a0)

> Saving game without name should result in "no name" game, now it's empty.

Thank you, will put it on my list. I also don't think it should be empty.

> Exiting options menu closes options window, it should go one lvl up instead I believe.

The arrow button on the left goes one level up and the exit button closes the window.

**Reply by rtrzebinski** [reply]
Thank you @nicymike for all your comments.

I added this to the wiki -> https://github.com/nicymike/serflings/wiki/Misc#localize-buildings

> The arrow button on the left goes one level up and the exit button closes the window.

There is no arrow button in the settings menu however. Super not important, but just saying :)

**Reply by nicymike** [reply]
If you are talking about the main page of the options window your are correct. This is because the options window is not connected to any other window like for example the window for game settings, although there is a button for options. Therefore there is no level up for the main page of options.

The game can only display one window at a time. I plan to change this later including moving a window to a different location. But for now the window for game settings will be closed if you click on options and the window for options will then be opened.
